manifests/init.pp

```diff
@@ -456,8 +456,7 @@
   }
 
   if fact('service_provider') == 'systemd' {
-    systemd::dropin_file { 'snmpd.conf':
-      unit    => 'snmpd.service',
+    systemd::unit_file { 'snmpd.service':
```

Comment:
Doing so keeps the old drop in in place which can create problems for upgrading users.

I also strongly dislike overriding service files if the distro already provides one. That's a bad pattern. Why have you chosen to do this? Ubuntu 20.04 also provides a base snmpd.

Reply:
Could add an ensure absent on the drop in? The reasoning behind it the package provided one does not ship systemd with an EnvironmentFile, which then makes the defaults obsolete?

Reply:
Then add the `EnvironmentFile` via a drop in? I think that should work. The reason it's not there is that `EnvironmentFile=/etc/sysconfig` (or default) is not how systemd files are generally deployed. Without them, a simple `systemctl cat $service` shows you the whole config.

Reply:
Pushed tested on 20.04